acceptance/lib/puppet/acceptance/i18n_utils.rb

```diff
@@ -22,14 +23,16 @@ def enable_locale_language(agent, language)
             end
           end
         end
-        return language_name(agent, language)
+        return language_name(agent, language, prefer_utf8: prefer_utf8)
       end
 
       # figure out the preferred language string for the requested language if the language is configured on the system
-      def language_name(agent, language)
+      def language_name(agent, language, prefer_utf8: true)
         step "PLATFORM #{agent['platform']}"
           on(agent, 'locale -a') do |locale_result|
-            ["#{language}.utf8", "#{language}.UTF-8", language].each do |lang|
+            langs = [language]
+            langs.unshift("#{language}.utf8", "#{language}.UTF-8") if prefer_utf8
```

Comment:
This is the only part where the `prefer_utf8` option is used, which does not affect the behavior of setting the locale. If you do not use this option, then you'll have
```
[language.utf8, language.UTF-8, language]
```
as the matching array. This option is set to false in the first part of the tests, which will make the array be
```
[language]
```

and if language does not exist, the method would return nil and you'd skip the test. My point is would there be a case where {language}.utf8, {language}.UTF-8 would exist but {language} wouldn't in this matching array? That would seem odd. If there isn't a case, then you don't need the `prefer_utf8` option.

Reply:
You're right that it's very unlikely a utf8 version of a locale would exist when the non-utf8 version didn't. And I guess even if there were only a utf-8 version, the test wouldn't fail, it just wouldn't duplicate the bug. I will remove this parameter to keep things simpler.